ddclient

```diff
@@ -2283,9 +2266,9 @@ sub get_ip {
     if (defined($ip)) {
         # no need to parse $reply
     } elsif ($ip = ipv4_match($reply)) {
-        $ip = un_zero_pad($ip);
+        # got an IPv4 address
     } elsif ($ip = ipv6_match($reply)) {
-        $ip = un_zero_pad($ip);
+        # got a IPv6 address
     } else {
         warning("found neither ipv4 nor ipv6 address");
     }
```

Comment:
Yes, nice.  I just wanted to resolve the merge conflict and was rather unhappy with how it looked.  I expected there was a better way to do it.  Thanks for the much better way.

Reply:
Done.  This does require that `$ip` is set to `undef` when first defined (which it is) and also I set the value of a manually entered ip address into `$reply` rather than into `$ip` so that we will validate (and remove any embedded leading zeros) any address provided.